docker/lib/dependabot/docker/update_checker.rb

```diff
@@ -9,6 +9,24 @@
 require "dependabot/docker/requirement"
 require "dependabot/docker/utils/credentials_finder"
 
+module DockerRegistry2
```

Comment:
Thoughts on adding a comment here explaining why we need to monkey patch?

Reply:
Yeah, good call 👍

Reply:
looks like it's also used from the file parser, extract the monkey patch and require it in both places?

Reply:
And actually, to be safe I'm going to override this on the instance instead of globally. We reuse the docker registry in another class as well, so this might have (but currently doesn't) unintended consequences

Reply:
Ah right, not sure I fully follow the impact on what we do in the file parser, will this work there too or not needed? Digest use in file parser: https://github.com/dependabot/dependabot-core/blob/577c4039d51e88e324f8be5e4de5ff8d8822ed96/docker/lib/dependabot/docker/file_parser.rb#L103

Reply:
seems this is used to figure out the "from" version of the dependency, not sure if this makes any difference?